common/src/main/java/io/armory/plugin/observability/promethus/PrometheusScrapeEndpoint.java

```diff
@@ -16,51 +16,52 @@
 
 package io.armory.plugin.observability.promethus;
 
+import io.prometheus.client.Collector;
 import io.prometheus.client.CollectorRegistry;
 import io.prometheus.client.exporter.common.TextFormat;
 import java.io.IOException;
 import java.io.StringWriter;
-import java.util.Optional;
-import java.util.Set;
-import lombok.extern.slf4j.Slf4j;
+import java.io.Writer;
+import java.util.Enumeration;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.lang.Nullable;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
 
 /**
- * Port of PrometheusScrapeEndpoint but rather than being an actuator endpoint that won't work with
- * plugins, it will be a rest controller bean that we can register.
+ * Port of PrometheusScrapeEndpoint but rather than being an web endpoint that won't work with
+ * plugins, it will be an endpoint
  *
  * <p>See:
  * https://github.com/spring-projects/spring-boot/blob/cd1baf18fe9ec71c11d7d131d6f1a417ec0c00e2/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/export/prometheus/PrometheusScrapeEndpoint.java
  */
-@Slf4j
-@RestController()
-@RequestMapping("/armory-observability/metrics")
-public class PrometheusScrapeController {
+// If you use WebEndpoint instead of Endpoint, the plugin throws class def not found error with PF4j
+// ¯\_(ツ)_/¯
+@Endpoint(id = "prometheus")
```

Comment:
Should we namespace this to something like `armory-prometheus` so that if/when the micrometer prom reg lib ends up on the classpath we don't have a collision in the future?

Reply:
Good thinking. Wouldn't hurt and perhaps makes sense with this repo being under Armory organization.

I wonder if Spinnaker is more likely to bring in the project agnostic CNCF sandbox [OpenMetrics](https://openmetrics.io/) versus Prometheus as such. Lots of projects in same position.

Reply:
It is my understanding Prometheus is turning into open metrics.
I've seen them used interchangeably.
Open Metrics is a WIP and not ratified yet.

Once open metrics has Micrometer support, I would be more than willing to at least add it to this plugin.
But for the time being Prometheus is sort of the closest thing you have to vendor-agnostic format because the vendors I've checked out (signalfx, new relic and datadog) have Prometheus scrappers.

Spinnaker already has a vendor-agnostic framework integrated via Micrometer, the idea is you chose the micrometer registry you want.

Reply:
Case in point: https://docs.newrelic.com/docs/integrations/prometheus-integrations/get-started/new-relic-prometheus-openmetrics-integration-kubernetes

☝️ New Relic calls their Prometheus scrapper an Open Metrics integration